compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -250,6 +302,8 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
 
         let get_abi = |this: &Self, instance_ty: Ty<'tcx>| match instance_ty.kind() {
```

Comment:
Hm... looking at FnAbi, I think what I should do here is remove the Abi comparison entirely and instead compare the `conv` field of `FnAbi`.

Reply:
That sounds right - also, FWIW, most of the time, the "calling convention" won't even be anything other than "default for C on that platform", since the `extern "Rust"` and `extern "C"` differences are all encoded in how arguments/returns are passed around.

Reply:
You may still want to emit a warning for "Rust" vs "non-Rust" - even when compatible by happenstance, relying on it is probably a bad idea.

Reply:
We're not really in the business of emitting warnings, but I see there are `Rust` and `C` variants of `Conv` so if we compare the `Conv` I assume we'd still catch Rust vs non-Rust mismatch.

Reply:
Ah, unfortunately the Abi is still needed to determine whether arguments need to be untupled... FnAbi doesn't contain that information. :/

Reply:
But other than that this works now, nice.